R/calcUvalueParameters.R

```diff
@@ -0,0 +1,567 @@
+
+  regionmapMessage <- toolGetMapping("regionmappingMessageIX.csv",
+                                     type = "regional",
+                                     where = "mredgebuildings")
+
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+
+  ## Process U-Value Data ====
+
+
+  ### Hotmaps Data =====
+
+  # U-Values per building vintage and building component (e.g. floor, windows, ...)
+  uvaluesHotmapsData <- hotmapsData %>%
+    filter(.data$variable %in% uvalueParameters,
+           !.data$building %in% buildingsToIgnore) %>%
+    mutate(variable = str_to_lower(word(.data$variable, -1, sep = "\\|"))) %>%
+    select(-"unit", -"model", -"scenario")
+
+  # Prepare building components
+  buildingCharacteristics <- buildingCharacteristics %>%
+    mutate(variable = recode(.data$variable, !!!varMapBuildCharacteristics)) %>%
+    rename("weight" = "value") %>%
+    select(-"unit", -"source", -"floorspace", -"component_area")
+
+  # Aggregate U-Values per building bage with weighted mean across component areas
+  uvaluesAggComp <- uvaluesHotmapsData %>%
+    left_join(buildingTypeMap, by = c("building" = "typHotmaps")) %>%
+    left_join(buildingCharacteristics, by = c("variable", "typBEAM" = "building")) %>%
+    group_by(across(all_of(c("region", "period", "bage", "building")))) %>%
+    reframe(value = sum(.data$weight * .data$value)) %>%
+    ungroup()
+
+  # Floor space as aggregation weights
+  floorspaceHotmaps <- hotmapsData %>%
+    filter(.data$variable %in% floorParameters,
+           !.data$building %in% buildingsToIgnore) %>%
+    rename("floorspace" = "value") %>%
+    select(-"scenario", -"model", -"unit", -"variable")
+
+  # Aggregate u-values across building bages and types
+  uvaluesHotmaps <- uvaluesAggComp %>%
+    filter(!is.na(.data$value)) %>%
+    left_join(floorspaceHotmaps, by = c("region", "period", "bage", "building")) %>%
+    group_by(across(all_of(c("region", "period")))) %>%
+    reframe(uvalue = sum(.data$value * .data$floorspace) / sum(.data$floorspace))
+
+
+
+  ### EU Building Observatory ====
+
+  # Floor space weights
+  floorspaceOdyssee <- odysseeData %>%
+    # rename variables
+    filter(.data$variable %in% names(varMapOdyssee)) %>%
+    mutate(variable = recode(.data$variable, !!!varMapOdyssee)) %>%
+    select("region", "period", "variable", "value") %>%
+
+    # calculate total residential floor space
+    pivot_wider(names_from = "variable", values_from = "value") %>%
+    mutate(floorRes = .data$nDwellings * .data$residentialAvg) %>%
+    select(-"nDwellings", -"residentialAvg") %>%
+
+    # calculate regional residential share
+    group_by(across(all_of(c("region", "period")))) %>%
+    mutate(floorShareRes = .data$floorRes / (.data$floorRes + .data$floorCom)) %>%
+    ungroup() %>%
+
+    # calculate global average shares per period
+    group_by(across(all_of("period"))) %>%
+    mutate(floorShareResGlo = mean(.data$floorShareRes, na.rm = TRUE)) %>%
```

Comment:
Try to avoid unweighted averaging if possible
```suggestion
    mutate(isComplete = !is.na(.data$floorRes) & !is.na(.data$floorCom),
           floorShareResGlo = sum(.data$floorRes[.data$isComplete]) / sum(.data$floorRes + .data$floorCom, na.rm = TRUE)) %>%
```